packages/design-system/src/components/SearchBar/SearchBar.js

```diff
@@ -0,0 +1,168 @@
+import * as React from 'react';
+import * as PropTypes from 'prop-types';
+import cx from 'classnames';
+import SearchIcon from 'react-material-icon-svg/dist/MagnifyIcon';
+import CloseIcon from 'react-material-icon-svg/dist/CloseIcon';
+import { debounce } from '@livechat/data-utils';
+
+import styles from './style.scss';
+import { Loader } from '../Loader';
+import { KeyCodes } from '../../constants/keyCodes';
+
+const baseClass = 'search-bar';
+const noop = () => {};
+
+class SearchBar extends React.PureComponent {
+  constructor(props) {
+    super(props);
+
+    this.inputRef = React.createRef();
+    this.onChangeDebounced = debounce(this.props.debounceInMs, value => {
```

Comment:
move it out from constructor and rename it to `debouncedOnChange` 
also if you use debounce you need to make sure that debounce is canceled before the component will unmount

Reply:
Why move it our from the constructor?